### PR TITLE
build and test workflow

### PR DESCRIPTION
**Pull request by eirsep**
Adds github workflows for the plugin

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/multi-node-test-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      # This step uses the checkout Github action: https://github.com/actions/checkout
+      - name: Checkout Branch
+        uses: actions/checkout@v2
+      # This step uses the setup-java Github action: https://github.com/actions/setup-java
+      - name: Set Up JDK 14
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
```
Comment (Bukhtawar) [inline .github/workflows/multi-node-test-workflow.yml:25]: can this be parameterized

File: .github/workflows/release-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
+name: Release workflow
+# This workflow is triggered on creating tags to master or a opendistro release branch
+on:
+  #  push:
+  #    tags:
+  #      - 'v*'
```
Comment (Bukhtawar) [inline .github/workflows/release-workflow.yml:6]: remove?

File: .github/workflows/release-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      # Creating artifact path as well as individual folders for rpm, zip and deb assets
+      - name: Create Artifact Path
+        run: |
+          mkdir -p asynchronous-search-artifacts
+          cp ./build/distributions/*.deb asynchronous-search-artifacts
+          cp ./build/distributions/*.deb asynchronous-search-artifacts_deb
+          cp ./build/distributions/*.rpm asynchronous-search-artifacts
+          cp ./build/distributions/*.rpm asynchronous-search-artifacts_rpm
+          cp ./build/distributions/*.zip asynchronous-search-artifacts
+          cp ./build/distributions/*.zip asynchronous-search-artifacts_zip
```
Comment (Bukhtawar) [inline .github/workflows/release-workflow.yml:43]: whats the standard artifact naming convention?

Comment (Bukhtawar) [reply .github/workflows/release-workflow.yml:43]: can we extract out the common?

File: .github/workflows/release-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+          echo "TAG_VERSION=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+      # AWS authentication
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_STAGING_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_STAGING_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
```
Comment (Bukhtawar) [inline .github/workflows/release-workflow.yml:51]: parameterize?

**Review by Bukhtawar — COMMENTED** [review]
Thanks for the changes Surya

File: .github/workflows/release-workflow.yml (hunk 3 above)
Comment (eirsep) [reply .github/workflows/release-workflow.yml]: > whats the standard artifact naming convention?

referring to https://quip-amazon.com/gmuHA2uDtWHV/ODFE-artifact-naming-convention

From release workflow run logs : 
s3://staging.artifacts.opendistroforelasticsearch.amazon.com/snapshots/elasticsearch-plugins/asynchronous-search/opendistro-asynchronous-search-1.11.1.0-build-18.zip
s3://staging.artifacts.opendistroforelasticsearch.amazon.com/snapshots/elasticsearch-plugins/asynchronous-search/opendistro-asynchronous-search-1.11.1.0-build-18.deb
s3://staging.artifacts.opendistroforelasticsearch.amazon.com/snapshots/elasticsearch-plugins/asynchronous-search/opendistro-asynchronous-search-1.11.1.0-build-18.rpm

Sreekar from infra team okayed the names. Had shared an earlier build with him.

File: .github/workflows/release-workflow.yml (hunk 2 above)
Comment (eirsep) [reply .github/workflows/release-workflow.yml]: removed on push event and uncommented tag event trigger now that action is tested.

**Review by Bukhtawar — APPROVED** [review]

**Review by anshul291995 — APPROVED** [review]

**Pull request merged**